internal/codegen/tpl_file.go

```diff
@@ -260,6 +264,78 @@ func (f *TplFile) MigrateTo(path string) (out string, err error) {
 	return "", nil
 }
 
+// MigrateFrom parses blocks from the provided file as if they
+// originated from this template. This is useful when you want to move a
+// file, but keep its contents. This should be called before any other
+// functions are called inside of the file.
+//
+// Note: Only blocks are persisted. Unlike [MigrateTo] the file is not
+// actually copied to disk.
+//
+// **Example `old.go`:**
+//
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	func main() {
+//	    fmt.Println("Hello from migrated code!")
+//	}
+//	## <</Stencil::Block>>
+//
+// **Example `new.go.tpl`:**
+//
+//	{{ file.MigrateFrom "old.go" -}}
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	{{ file.Block "main-function" }}
+//	## <</Stencil::Block>>
+//
+// **Example `new.go`:**
+//
+//	package main
+//
+//	import "fmt"
+//
+//	## <<Stencil::Block(main-function)>>
+//	func main() {
+//	    fmt.Println("Hello from migrated code!")
+//	}
+//	## <</Stencil::Block>>
+//
+// Result: The block content from old.go is preserved in new.go
+func (f *TplFile) MigrateFrom(path string) (out string, err error) {
+	// Only load blocks from the old file if the new file doesn't exist
+	if _, err := osfs.Default.Stat(f.f.path); err != nil {
+		blocks, err := parseBlocks(path, f.t)
+		if err != nil {
+			return "", fmt.Errorf("MigrateFrom: failed to parse blocks from %s: %w", path, err)
+		}
+
+		f.f.blocks = blocks
+	} else {
+		f.log.Debug("MigrateFrom: not loading blocks due to destination file existing")
+	}
+
+	// If the old file exists, delete it
+	if _, err := osfs.Default.Stat(path); err == nil {
```

Comment:
i guess you have to pick how you want teh edge case of both new and old exist, what to do?  it feels sitll like you want to both read blocks from that and delete it, even though the new file exists already, but i dunno, i could go either way there.

Reply:
Opted to use `old.go` over new.